src/GafferScene/Instancer.cpp

```diff
@@ -464,16 +464,110 @@ class Instancer::EngineData : public Data
 					auto ins = m_idsToPointIndices.try_emplace( id, i );
 					if( !ins.second )
 					{
+						// We have multiple indices trying to use this id.
 						if( !omitDuplicateIds )
 						{
 							throw IECore::Exception( fmt::format( "Instance id \"{}\" is duplicated at index {} and {}. This probably indicates invalid source data, if you want to hack around it, you can set \"omitDuplicateIds\".", id, m_idsToPointIndices[id], i ) );
 						}
 
-						// Invalidate the existing entry in the m_idsToPointIndices map - since we can't assign
-						// a consistent point index to this id, we omit all point indices that try to use this
-						// id
-						ins.first->second = std::numeric_limits<size_t>::max();
-						m_hasDuplicates = true;
+						if( !m_indicesInactive.size() )
+						{
+							m_indicesInactive.resize( numPoints(), false );
+						}
+
+						// If we're omitting duplicate ids, then we need to omit both the current index, and
+						// the index that first tried to use this id.
+						m_indicesInactive[ i ] = true;
+						m_indicesInactive[ ins.first->second ] = true;
+					}
+				}
+			}
+
+			std::vector<std::string> inactiveIdVarNames;
+			IECore::StringAlgo::tokenize( inactiveIds, ' ', inactiveIdVarNames );
+			for( std::string &inactiveIdVarName : inactiveIdVarNames )
+			{
+				if( m_primitive->variables.find( inactiveIdVarName ) == m_primitive->variables.end() )
+				{
+					continue;
+				}
+
+				const PrimitiveVariable *vertexInactiveVar = findVertexVariable( m_primitive.get(), inactiveIdVarName );
+				if( vertexInactiveVar )
+				{
+					if( const auto *vertexInactiveData = IECore::runTimeCast<BoolVectorData>( vertexInactiveVar->data.get() ) )
```

Comment:
DeletePoints accepts IntVectorData here as well as BoolVectorData. It'd be good to support that here too, particularly because we can't output BoolVectorData from OSLObject.

Reply:
Addressed.